runtime-kanto/src/app_deployment/deploy_vehicleapp.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2023 Robert Bosch GmbH
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import subprocess
+import sys
+from io import TextIOWrapper
+
+from yaspin import yaspin
+
+from velocitas_lib import create_log_file, get_app_manifest
+from velocitas_lib.docker import (
+    is_docker_image_build_locally,
+    push_docker_image_to_registry,
+)
+from velocitas_lib.services import get_service_port
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "..", "app_deployment"))
+from build_vehicleapp import build_vehicleapp  # noqa: E402
+
+
+def is_vehicleapp_installed(
+    app_name: str,
+    log_output: TextIOWrapper | int = subprocess.DEVNULL,
+) -> bool:
+    """Return whether the runtime is installed or not.
+
+    Args:
+        app_name (str): App name
+        log_output (TextIOWrapper | int): Logfile to write or DEVNULL by default.
+    """
+    return (
+        subprocess.call(
+            ["kanto-cm", "get", "-n", app_name],
+            stdout=log_output,
+            stderr=log_output,
+        )
+        == 0
+    )
+
+
+def remove_vehicleapp(
+    app_name: str, log_output: TextIOWrapper | int = subprocess.DEVNULL
+):
+    """Uninstall VehicleApp container
+
+    Args:
+        app_name (str): App name to remove container for
+        log_output (TextIOWrapper | int): Logfile to write or DEVNULL by default.
+    """
+    subprocess.call(
+        ["kanto-cm", "remove", "-f", "-n", app_name],
+        stdout=log_output,
+        stderr=log_output,
+    )
+
+
+def create_container(
+    app_name: str, log_output: TextIOWrapper | int = subprocess.DEVNULL
+):
+    """Create kanto container
+
+    Args:
+        app_name (str): App name for container creation
+        log_output (TextIOWrapper | int): Logfile to write or DEVNULL by default.
+    """
+    middleware_type = "native"
```

Comment:
since we want to be as close to the hardware as possible, we wanted to only use kanto without dapr. So for now and most probably forever Kanto will only be used with native middleware.